rx-extensions/src/main/java/com/appunite/rx/subjects/CacheSubject.java

```diff
@@ -67,13 +72,15 @@
         };
     }
 
-    private CacheSubject(@Nonnull final CacheCreator<T> cacheCreator, DelegateOnSubscribe<T> delegateOnSubscribe) {
+    private CacheSubject(@Nonnull final CacheCreator<T> cacheCreator,
+                         @Nonnull DelegateOnSubscribe<T> delegateOnSubscribe,
+                         final boolean skipFirstNull) {
         super(delegateOnSubscribe);
         delegateOnSubscribe.setDelegate(new OnSubscribe<T>() {
             @Override
             public void call(final Subscriber<? super T> child) {
                 final T t = cacheCreator.readFromCache();
-                if (t != null) {
+                if (!skipFirstNull || t != null) {
```

Comment:
You can look on test: https://github.com/jacek-marchwicki/rx-java-extensions/pull/54/files#diff-e51996ba7efd80fa0b507c160d1dc9d1R125 thet were exist